lib/dal/src/workspace_snapshot/graph/validator/connections.rs

```diff
@@ -56,14 +57,34 @@ pub fn connection_migrations_with_text(
 
 pub fn connection_migrations(
     graph: &impl std::ops::Deref<Target = WorkspaceSnapshotGraphVCurrent>,
+    inferred_connections: impl IntoIterator<Item = SocketConnection>,
 ) -> Vec<ConnectionMigration> {
     let graph = graph.deref();
-    let mut migrations = graph
+    let inferred_connection_migrations =
```

Comment:
We may want to move this onto the "slow_rt" if it is going to be expensive on large graphs. (Otherwise it will tie up the main executor)

Reply:
Sure :) It mainly depends on the definition of expensive: this will go through all socket connections in the graph, but does a constant-time burst of graph traversal on each one to determine its migrateabiliity. I'd be very surprised if it took more than single-digit milliseconds for 1000 connections.

I don't mind at *all* throwing this on slow_rt, but I'm asking to get a gauge: what's the feeling on how long a piece of sync code should run before we'd like it off the reactor?